account_move_template/tests/test_account_move_template_options.py

```diff
@@ -0,0 +1,294 @@
+                "journal_id": self.journal.id,
+                "line_ids": [
+                    Command.create(ar_line),
+                    Command.create(income_line1),
+                    Command.create(income_line2),
+                ],
+            }
+        )
+
+    def test_move_template_normal(self):
+        """Test normal case, input amount 300"""
+        with Form(self.env["account.move.template.run"]) as f:
+            f.template_id = self.move_template
+        template_run = f.save()
+        template_run.load_lines()
+        template_run.line_ids[0].amount = 300
+        res = template_run.generate_move()
+        move = self.Move.browse(res["res_id"])
+        self.assertRecordValues(
+            move.line_ids.sorted("credit"),
+            [
+                {"account_id": self.ar_account_id.id, "credit": 0.0, "debit": 300.0},
+                {
+                    "account_id": self.income_account_id.id,
+                    "credit": 100.0,
+                    "debit": 0.0,
+                },
+                {
+                    "account_id": self.income_account_id.id,
+                    "credit": 200.0,
+                    "debit": 0.0,
+                },
+            ],
+        )
+
+    def test_move_template_optional(self):
+        """Test optional case, input amount -300, expect optional account"""
+        with Form(self.env["account.move.template.run"]) as f:
+            f.template_id = self.move_template
+        template_run = f.save()
+        template_run.load_lines()
+        template_run.line_ids[0].amount = -300  # Negative amount
+        res = template_run.generate_move()
+        move = self.Move.browse(res["res_id"])
+        self.assertRecordValues(
+            move.line_ids.sorted("debit"),
+            [
+                {"account_id": self.ap_account_id.id, "credit": 300.0, "debit": 0.0},
+                {
+                    "account_id": self.expense_account_id.id,
+                    "credit": 0.0,
+                    "debit": 100.0,
+                },
+                {
+                    "account_id": self.expense_account_id.id,
+                    "credit": 0.0,
+                    "debit": 200.0,
+                },
+            ],
+        )
+
+    def test_move_template_overwrite(self):
+        """Test case overwrite, amount = 3000, no need to manual input"""
+        # Test for error when debit is not a valid field
+        with Form(self.env["account.move.template.run"]) as f:
+            f.template_id = self.move_template
+            f.overwrite = str(
+                {
+                    "L0": {
+                        "partner_id": self.partners[0].id,
+                        "amount": 3000,
+                        "debit": 3000,
+                    },
+                }
+            )
+        template_run = f.save()
+        msg_error = re.escape(
+            "Valid fields to overwrite are ['partner_id', 'amount', 'name', 'date_maturity']"
```

Comment:
Maybe the following to dont need `re.escape`

```suggestion
            "Valid fields to overwrite are .'partner_id', 'amount', 'name', 'date_maturity'"
```

or `.*`

Reply:
For this case, I used the text "overwrite are .'partner_id', 'amount', 'name', 'date_maturity'" to avoid the extra lines, this because the 88 character limit was reached.